src/gt4py/backend/module_generator.py

```diff
@@ -110,28 +111,29 @@ def make_args_data_from_gtir(pipeline: GtirPipeline) -> ModuleData:
     referenced_field_params = {
         param.name for param in node.params if isinstance(param, gtir.FieldDecl)
     }
-    for name in referenced_field_params:
+    for name in sorted(referenced_field_params):
         data.field_info[name] = FieldInfo(
```

Comment:
Was there a reason this needed to be sorted? It seems to me like `field_info` should be unordered.

Reply:
In principle sorting is not needed, but the generated code wouldn't be 100% reproducible since some parts can be in different order, which makes comparing snippets of code harder.

Reply:
It looks like this is a sorted loop, but it's assigning values into an unsorted container. Maybe I'm misunderstanding what is going on here, but it seems like the loop order shouldn't matter.

Reply:
`data.field_info` is a `dict`, and `dict`s are guaranteed to be ordered in python since 3.7.

Reply:
Ah, ok you're counting on that. I was still under the impression that it was a implementation detail, but it looks like that did become an official language feature in 3.7, and it looks like `setup.py` does enforce `python >= 3.8`, so we're all good!